internal/util/typeutil/schema.go

```diff
@@ -395,6 +397,46 @@ func AppendFieldData(dst []*schemapb.FieldData, src []*schemapb.FieldData, idx i
 	}
 }
 
+// DeleteFieldData delete fields data from src to dst
+func DeleteFieldData(dst []*schemapb.FieldData, src []*schemapb.FieldData, idx int64) {
+	for i, fieldData := range src {
+		switch fieldType := fieldData.Field.(type) {
+		case *schemapb.FieldData_Scalars:
+			dstScalar := dst[i].GetScalars()
+			switch fieldType.Scalars.Data.(type) {
+			case *schemapb.ScalarField_BoolData:
+				dstScalar.GetBoolData().Data = dstScalar.GetBoolData().Data[:len(dstScalar.GetBoolData().Data)-1]
+			case *schemapb.ScalarField_IntData:
+				dstScalar.GetIntData().Data = dstScalar.GetIntData().Data[:len(dstScalar.GetIntData().Data)-1]
+			case *schemapb.ScalarField_LongData:
+				dstScalar.GetLongData().Data = dstScalar.GetLongData().Data[:len(dstScalar.GetLongData().Data)-1]
+			case *schemapb.ScalarField_FloatData:
+				dstScalar.GetFloatData().Data = dstScalar.GetFloatData().Data[:len(dstScalar.GetFloatData().Data)-1]
+			case *schemapb.ScalarField_DoubleData:
+				dstScalar.GetDoubleData().Data = dstScalar.GetDoubleData().Data[:len(dstScalar.GetDoubleData().Data)-1]
+			case *schemapb.ScalarField_StringData:
+				dstScalar.GetStringData().Data = dstScalar.GetStringData().Data[:len(dstScalar.GetStringData().Data)-1]
+			default:
+				log.Error("wrong field type added", zap.String("field type", fieldData.Type.String()))
+			}
+		case *schemapb.FieldData_Vectors:
+			dim := fieldType.Vectors.Dim
+			dstVector := dst[i].GetVectors()
+			switch srcVector := fieldType.Vectors.Data.(type) {
+			case *schemapb.VectorField_BinaryVector:
+				dstBinaryVector := dstVector.Data.(*schemapb.VectorField_BinaryVector)
+				srcToDelete := srcVector.BinaryVector[idx*(dim/8) : (idx+1)*(dim/8)]
```

Comment:
`idx` is not needed in here, the last binary vector has a length of  `(dim/8)`, the last float vector has a length of `dim`, just remove the last `dim/8` or `dim` is alright,  `idx` is to retrieve data from `srcs`.